[Concept Entry] Sklearn multioutput-regression (#5820)
Codecademy/docs: content/sklearn/concepts/multioutput-regression/multioutput-regression.md

---
Title: 'Multioutput Regression'
Description: 'Multioutput regression is a type of regression task where the model predicts multiple dependent variables (outputs) simultaneously for each input.'
Subjects:
  - 'Data Science'
  - 'Machine Learning'
Tags:
  - 'Classification'
  - 'Multitask Learning'
  - 'MultiTaskLasso'
  - 'Scikit-learn'
CatalogContent:
  - 'learn-python-3'
  - 'paths/data-science'
---

> In sklearn, **Multioutput Regression** is a type of regression task where the model predicts multiple dependent variables (outputs) simultaneously for each input, allowing for the modeling of relationships between multiple target variables and the features, which can improve prediction accuracy when outputs are correlated.

Reviewer: Add a link to sklearn docs entry.

```suggestion
In [sklearn](https://www.codecademy.com/resources/docs/sklearn), **Multioutput Regression** is a type of regression task where the model predicts multiple dependent variables (outputs) simultaneously for each input, allowing for the modeling of relationships between multiple target variables and the features, which can improve prediction accuracy when outputs are correlated.
```

> This can be achieved using the `MultiOutputRegressor` class, which wraps a single-output regressor (like `LinearRegression` or `DecisionTreeRegressor`) and fits a separate model for each target variable. The model then predicts all outputs at once for each input, treating each target independently.

Reviewer: We can add links to Linear Regression and Decision Tree Regressor here as well:

```suggestion
This can be achieved using the `MultiOutputRegressor` class, which wraps a single-output regressor (like [`LinearRegression`](https://www.codecademy.com/resources/docs/sklearn/linear-regression-analysis) or [`DecisionTreeRegressor`](https://www.codecademy.com/resources/docs/sklearn/decision-trees)) and fits a separate model for each target variable. The model then predicts all outputs at once for each input, treating each target independently.
```

## Syntax

```pseudo

> sklearn.multioutput.MultiOutputRegressor(estimator, n_jobs=None)

Reviewer: In Scikit-learn, `MultiOutputRegressor` is located in the `sklearn.multioutput` module. However, the `sklearn.multioutput` part is just the module path, and `MultiOutputRegressor` needs to be imported explicitly from that module. It can't be used as in current syntax.

So, the correct way to use it would be:

```suggestion
from sklearn.multioutput import MultiOutputRegressor

multi_output_regressor = MultiOutputRegressor(estimator, n_jobs=None)
```

```

- `estimator`: The base regressor that is used to fit each target independently. This can be any regression model that supports single-output regression (e.g., `LinearRegression`, `DecisionTreeRegressor`, etc.).
- `n_jobs`: The number of jobs to run in parallel for fitting the models.
  - If `None`, it defaults to 1 (single-threaded).
  - If `-1`, it uses all available processors.
  - If `int > 1`, it uses that many processors for parallel computation.

## Example

> In the following example, a multi-output regression model is trained using `MultiOutputRegressor` with `LinearRegression` as the base estimator to predict two target variables from a dataset with 100 samples and 10 features:

Reviewer: ```suggestion
In the following example, a multi-output regression model is trained using `MultiOutputRegressor` with `LinearRegression` as the base `estimator` to predict two target variables from a dataset with 100 samples and 10 features:
```

 [ 130.65561152   66.64815668]
 [ -85.31407057 -168.81530534]
 [  -7.2468998    73.28377393]
 [  22.33697872  145.21764028]
 [-120.51168929 -342.963189  ]
 [ 121.12613888   65.01661617]
 [ 124.10868505  354.92584718]
 [-147.66348249 -294.81859794]
 [  61.14523063   60.52117341]
 [-126.37893383 -334.70135616]
 [-111.77099591  -81.93814188]
 [-109.83747752 -237.97526597]
 [   8.00415806   91.38676316]
 [ -26.37947013   36.09839868]
 [ 106.36699275  130.83993429]
 [  69.06778835  125.59665375]
 [ 134.03028548  319.28586998]
 [ 130.75716498   15.34231243]
 [ -86.46672131 -139.61281879]
 [  -7.33734137 -226.69848199]
 [ 199.71269604  357.97063185]
 [ 100.94948846  -32.96835461]
 [-257.05342439 -386.6851282 ]
 [ -99.42556327 -108.57915827]
 [ 224.41784227  425.50742575]
 [-269.92957188 -202.28685621]
 [-109.21584421 -225.03205094]
 [-118.45089966 -420.99745962]
 [ -29.83876402   19.58063146]
 [  95.06986687   70.5609531 ]
 [  41.32888453    4.51642366]
 [ -10.61243193  289.0884239 ]
 [  73.11234969  158.84947994]
 [  10.45019796  260.51876186]
 [-226.04884764 -372.71451196]
 [ -17.30979575 -146.3735002 ]
 [ -13.07113033  -42.21748842]
 [ -59.54942557 -102.03957313]]
```

> > **Note:** The output will vary each time the code is run unless a fixed `random_state` is set in `make_regression()` to ensure reproducibility.

Reviewer: The phrasing can be made slightly clearer regarding the role of `random_state`:

```suggestion
> **Note:** The output will vary each time the code is run unless a fixed `random_state` is set in `make_regression()`, ensuring reproducibility as shown in the example.
```
